### PR TITLE
Example fix

### PR DESCRIPTION
**Pull request by Boris-Plato**
Related to https://github.com/rabbitmq/amqp091-go/issues/249

### CODE REVIEW

File: _examples/pubsub/pubsub.go [hunk 1]
```diff
@@ -117,6 +118,7 @@ func publish(sessions chan chan session, messages <-chan message) {
 			select {
 			case confirmed, ok := <-confirm:
 				if !ok {
+					pub.Close()
```
Comment (Gsantomaggio) [inline _examples/pubsub/pubsub.go:121]: Won't it be closed anyway here? https://github.com/rabbitmq/amqp091-go/blob/main/_examples/pubsub/pubsub.go#L135

Comment (Zerpet) [reply _examples/pubsub/pubsub.go:121]: I don't think so, because the `break Publish` exits the `for` loop.

**Review by Zerpet — COMMENTED** [review]
It's looking good. I'm going to verify manually before merging.

**Review by Zerpet — APPROVED** [review]
Thank you!

**Pull request merged**